include/xsimd/arch/xsimd_avx2.hpp

```diff
@@ -942,20 +940,33 @@ namespace xsimd
         template <class A, uint32_t V0, uint32_t V1, uint32_t V2, uint32_t V3, uint32_t V4, uint32_t V5, uint32_t V6, uint32_t V7>
         XSIMD_INLINE batch<float, A> swizzle(batch<float, A> const& self, batch_constant<uint32_t, A, V0, V1, V2, V3, V4, V5, V6, V7> mask, requires_arch<avx2>) noexcept
         {
-            return _mm256_permutevar8x32_ps(self, mask.as_batch());
+            XSIMD_IF_CONSTEXPR(detail::is_all_different(mask) && !detail::is_identity(mask))
+            {
+                // The intrinsic does NOT allow to copy the same element of the source vector to more than one element of the destination vector.
+                // one-shot 8-lane permute
```

Comment:
Was this tested concretely? According to the [intel ISA reference for VPERMPS](https://www.intel.com/content/dam/develop/external/us/en/documents/319433-024-697869.pdf), 
> this instruction permits a doubleword in the source operand to be copied to more than one location in the destination operand

Reply:
There was a webpage once but intel changed their website and I cannot find it anymore

Reply:
I would say the Intel ISA reference is more authoritative than the intrinsics documentation.

Success on Godbolt only tells you that the particular CPU allows duplicate values, not that it's allowed everywhere.

Reply:
And Intel's [current intrinsincs doc](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=permutevar&ig_expand=4953,4954,4953,4954&techs=AVX_ALL&cats=Swizzle) does not say anything about disallowing duplicate values.

Reply:
(@AntoinePRV you could be interested in this discussion)

Reply:
I could try to, but I'm not a xsimd developer (I just encountered this by chance when grepping for xsimd's support for shuffling).